src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,177 @@
+                        var container = semanticModel.GetTypeInfo(expression, cancellationToken).Type;
+                        if (container is { })
+                        {
+                            var allMembers = container.GetMembers();
+                            var allExplicitConversions = from m in allMembers.OfType<IMethodSymbol>()
+                                                         where
+                                                             m.IsConversion() && // MethodKind.Conversion
+                                                             m.Name == WellKnownMemberNames.ExplicitConversionName && // op_Explicit
+                                                             m.Parameters[0].Type == container // Convert from container type to other type
+                                                         select SymbolCompletionItem.CreateWithSymbolId(
+                                                             displayText: $"({m.ReturnType.ToMinimalDisplayString(semanticModel, position)})", // The type to convert to
+                                                             symbols: ImmutableList.Create(m),
+                                                             rules: CompletionItemRules.Default,
+                                                             contextPosition: position,
+                                                             properties: CreateCompletionHandlerProperty(CompletionHandlerConversion));
+                            var indexer = from p in allMembers.OfType<IPropertySymbol>()
+                                          where p.IsIndexer
+                                          select SymbolCompletionItem.CreateWithSymbolId(
+                                              displayText: $"[{string.Join(", ", p.Parameters.Select(p => p.Type.ToMinimalDisplayString(semanticModel, position)))}]", // The type to convert to
+                                              symbols: ImmutableList.Create(p),
+                                              rules: CompletionItemRules.Default,
+                                              contextPosition: position,
+                                              properties: CreateCompletionHandlerProperty(CompletionHandlerIndexer));
+                            context.AddItems(allExplicitConversions.Union(indexer));
+                        }
+                    }
+                }
+            }
+        }
+
+        private static ExpressionSyntax? GetExpressionOfInvocation(SyntaxToken dotToken)
+        {
+            Debug.Assert(dotToken.IsKind(SyntaxKind.DotToken));
+            return dotToken.Parent switch
+            {
+                MemberAccessExpressionSyntax memberAccess => memberAccess.Expression,
+                MemberBindingExpressionSyntax { Parent: ConditionalAccessExpressionSyntax conditionalAccess } _ => conditionalAccess.Expression,
```

Comment:
this won't work if you have `a?.$$b.c()`

Reply:
You are absolutely right! The conditionals killed my current solution. I tried some tree manipulation approaches but that was a dead end. I needed to resort to text manipulation  (see 594ea5c8).